sfge/src/main/java/com/salesforce/graph/build/ConstructorUtil.java

```diff
@@ -0,0 +1,168 @@
+package com.salesforce.graph.build;
+
+import com.salesforce.apex.jorje.ASTConstants.NodeType;
+import com.salesforce.apex.jorje.JorjeNode;
+import com.salesforce.graph.Schema;
+import java.util.List;
+import java.util.Optional;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+
+/** Handles the creation of synthetic vertices to allow graceful expansion of constructors. */
+public final class ConstructorUtil {
```

Comment:
*Utils classes generally end up collecting debts.   especially when it has numerous statics.  I have no specific suggestion here.  this is just a heads up.  static classes do not lend themsleves easily for unit testing too.  there are way to mitigate this by using Signleton pattern